plugins/inputs/supervisor/supervisor.go

```diff
@@ -0,0 +1,170 @@
+package supervisor
+
+import (
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/kolo/xmlrpc"
+	"net/url"
+)
+
+type Supervisor struct {
+	Log telegraf.Logger `toml:"-"`
+
+	Server         string `toml:"url"`
+	PidGather      bool   `toml:"gather_pid"`
+	ExitCodeGather bool   `toml:"gather_exit_code"`
+	UseIdentTag    bool   `toml:"use_identification_tag"`
+
+	Status SupervisorInfo
+}
+
+type ProcessInfo struct {
```

Comment:
I think we do not need to export this structure.
```suggestion
type processInfo struct {
```

Reply:
Will be included in next commit